<!doctype html><html lang=en dir=ltr class=scroll-smooth data-default-appearance=light data-auto-appearance=true><head><meta charset=utf-8><meta http-equiv=content-language content="en"><meta name=viewport content="width=device-width,initial-scale=1"><meta name=theme-color content="rgb(255,255,255)"><meta http-equiv=x-ua-compatible content="ie=edge"><title>Bi-Weekly news #13 and #14 &#183; Martin PROKS</title><meta name=title content="Bi-Weekly news #13 and #14 &#183; Martin PROKS"><meta name=description content="News # Richard Stallman is Back!"><link rel=canonical href=/weekly/weekly-news-13-14/><link type=text/css rel=stylesheet href=/css/main.bundle.min.17fc506b1c26b1bbc9a782ffffc510d9ecbfef39adb801fd6c673c614f713ffe.css integrity="sha256-F/xQaxwmsbvJp4L//8UQ2ey/7zmtuAH9bGc8YU9xP/4="><script type=text/javascript src=/js/appearance.min.022d0ebc3b46a335eb1c7ef79b7f2de143d7cd5156d433638592ef1ce5f8554e.js integrity="sha256-Ai0OvDtGozXrHH73m38t4UPXzVFW1DNjhZLvHOX4VU4="></script>
<script defer type=text/javascript id=script-bundle src=/js/main.bundle.min.75206d23ef83f4908b2bdd2317bf6ddff399e9173a16fff5451c40b8e857cfa8.js integrity="sha256-dSBtI++D9JCLK90jF79t3/OZ6Rc6Fv/1RRxAuOhXz6g=" data-copy=Copy data-copied=Copied></script>
<link rel=apple-touch-icon sizes=180x180 href=/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=/favicon-16x16.png><link rel=manifest href=/site.webmanifest><meta property="og:title" content="Bi-Weekly news #13 and #14"><meta property="og:description" content="News # Richard Stallman is Back!"><meta property="og:type" content="article"><meta property="og:url" content="/weekly/weekly-news-13-14/"><meta property="article:section" content="weekly"><meta property="article:published_time" content="2021-04-04T00:00:00+00:00"><meta property="article:modified_time" content="2021-04-04T00:00:00+00:00"><meta name=twitter:card content="summary"><meta name=twitter:title content="Bi-Weekly news #13 and #14"><meta name=twitter:description content="News # Richard Stallman is Back!"><script type=application/ld+json>[{"@context":"https://schema.org","@type":"Article","articleSection":"Weeklies","name":"Bi-Weekly news #13 and #14","headline":"Bi-Weekly news #13 and #14","abstract":"News # Richard Stallman is Back!","inLanguage":"en","url":"\/weekly\/weekly-news-13-14\/","author":{"@type":"Person","name":"Martin Proks"},"copyrightYear":"2021","dateCreated":"2021-04-04T00:00:00\u002b00:00","datePublished":"2021-04-04T00:00:00\u002b00:00","dateModified":"2021-04-04T00:00:00\u002b00:00","keywords":["weekly"],"mainEntityOfPage":"true","wordCount":"1753"}]</script><meta name=author content="Martin Proks"><link href=https://twitter.com/mproksik rel=me><link href=https://linkedin.com/in/martinproks rel=me><link href=https://github.com/matq007 rel=me><link href=https://orcid.org/0000-0002-8178-3128 rel=me><link href=https://www.researchgate.net/profile/Martin-Proks rel=me></head><body class="flex flex-col h-screen px-6 m-auto text-lg leading-7 max-w-7xl bg-neutral text-neutral-900 dark:bg-neutral-800 dark:text-neutral sm:px-14 md:px-24 lg:px-32"><div id=the-top class="absolute flex self-center"><a class="px-3 py-1 text-sm -translate-y-8 rounded-b-lg bg-primary-200 focus:translate-y-0 dark:bg-neutral-600" href=#main-content><span class="font-bold pe-2 text-primary-600 dark:text-primary-400">&darr;</span>Skip to main content</a></div><header class="py-6 font-semibold text-neutral-900 dark:text-neutral print:hidden sm:py-10"><nav class="flex items-start justify-between sm:items-center"><div class="flex flex-row items-center"><a class="decoration-primary-500 hover:underline hover:decoration-2 hover:underline-offset-2" rel=me href=/>Martin PROKS</a></div><ul class="flex list-none flex-col ltr:text-right rtl:text-left sm:flex-row"><li class="group mb-1 sm:mb-0 sm:me-7 sm:last:me-0.5"><a href=/posts/ title><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Blog</span></a></li><li class="group mb-1 sm:mb-0 sm:me-7 sm:last:me-0.5"><a href=/projects/ title=Projects><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Projects</span></a></li><li class="group mb-1 sm:mb-0 sm:me-7 sm:last:me-0.5"><a href=/weekly/ title=Weeklies><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Weekly</span></a></li><li class="group mb-1 sm:mb-0 sm:me-7 sm:last:me-0.5"><a href=https://github.com/matq007/ title target=_blank><span class="group-dark:hover:text-primary-400 transition-colors group-hover:text-primary-600"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path fill="currentcolor" d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6.0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6.0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3.0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1.0-6.2-.3-40.4-.3-61.4.0.0-70 15-84.7-29.8.0.0-11.4-29.1-27.8-36.6.0.0-22.9-15.7 1.6-15.4.0.0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5.0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9.0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4.0 33.7-.3 75.4-.3 83.6.0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6.0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9.0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg></span></span></a></li><li class="group mb-1 sm:mb-0 sm:me-7 sm:last:me-0.5"><button id=search-button-1 title="Search (/)">
<span class="group-dark:hover:text-primary-400 transition-colors group-hover:text-primary-600"><span class="relative inline-block align-text-bottom px-1 icon"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="search" class="svg-inline--fa fa-search fa-w-16" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M505 442.7 405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9.0 208 0S0 93.1.0 208s93.1 208 208 208c48.3.0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9.0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7.0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7.0 128 57.2 128 128 0 70.7-57.2 128-128 128z"/></svg></span></span><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2"></span></button></li></ul></nav></header><div class="relative flex flex-col grow"><main id=main-content class=grow><article><header class=max-w-prose><h1 class="mt-0 text-4xl font-extrabold text-neutral-900 dark:text-neutral">Bi-Weekly news #13 and #14</h1><div class="mt-8 mb-12 text-base text-neutral-500 dark:text-neutral-400 print:hidden"><div class="flex flex-row flex-wrap items-center"><time datetime="2021-04-04 00:00:00 +0000 UTC">4 April 2021</time><span class="px-2 text-primary-500">&#183;</span><span title="Reading time">9 mins</span></div><div class="my-1 text-xs leading-relaxed text-neutral-500 dark:text-neutral-400"><a href=/tags/weekly/ class="rounded-md border border-neutral-200 px-1 py-[1px] hover:border-primary-300 hover:text-primary-700 dark:border-neutral-600 dark:hover:border-primary-600 dark:hover:text-primary-400">weekly</a></div></div></header><section class="flex flex-col max-w-full mt-0 prose dark:prose-invert lg:flex-row"><div class="order-first px-0 lg:order-last lg:max-w-xs lg:ps-8"><div class="toc pe-5 print:hidden lg:sticky lg:top-10"><details open class="-ms-5 mt-0 overflow-hidden rounded-lg ps-5"><summary class="-ms-5 block cursor-pointer bg-neutral-100 py-1 ps-5 text-lg font-semibold text-neutral-800 dark:bg-neutral-700 dark:text-neutral-100 lg:hidden">Table of Contents</summary><div class="-ms-5 border-s border-dotted border-neutral-300 py-2 ps-5 dark:border-neutral-600"><nav id=TableOfContents><ul><li><a href=#i-classfas-fa-bullhorni-news><i class="fas fa-bullhorn"></i> News</a></li><li><a href=#i-classfas-fa-dnai-sciencebioinformatics><i class="fas fa-dna"></i> Science/Bioinformatics</a><ul><li><a href=#abel-prize-celebrates-union-of-mathematics-and-computer-sciencehttpswwwnaturecomarticlesd41586-021-00694-9><a href=https://www.nature.com/articles/d41586-021-00694-9>Abel Prize celebrates union of mathematics and computer science</a></a></li><li><a href=#an-overview-of-next-generation-sequencinghttpswwwtechnologynetworkscomgenomicsarticlesan-overview-of-next-generation-sequencing-346532><a href=https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532>An Overview of Next-Generation Sequencing</a></a></li><li><a href=#robust-integration-of-multiple-single-cell-rna-sequencing-datasets-using-a-single-reference-spacehttpswwwnaturecomarticless41587-021-00859-x><a href=https://www.nature.com/articles/s41587-021-00859-x>Robust integration of multiple single-cell RNA sequencing datasets using a single reference space</a></a></li><li><a href=#integrated-intra-and-intercellular-signaling-knowledge-for-multicellular-omics-analysishttpswwwembopressorgdoifull1015252msb20209923><a href=https://www.embopress.org/doi/full/10.15252/msb.20209923>Integrated intra‚Äê and intercellular signaling knowledge for multicellular omics analysis</a></a></li><li><a href=#sepal-identifying-transcript-profiles-with-spatial-patterns-by-diffusion-based-modelinghttpsacademicoupcombioinformaticsadvance-articledoi101093bioinformaticsbtab1646168120><a href=https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btab164/6168120>sepal: identifying transcript profiles with spatial patterns by diffusion-based modeling</a></a></li><li><a href=#user-friendly-scalable-tools-and-workflows-for-single-cell-rna-seq-analysishttpswwwnaturecomarticless41592-021-01102-w><a href=https://www.nature.com/articles/s41592-021-01102-w>User-friendly, scalable tools and workflows for single-cell RNA-seq analysis</a></a></li><li><a href=#enabling-highthroughput-biology-with-flexible-opensource-automationhttpswwwembopressorgdoifull1015252msb20209942><a href=https://www.embopress.org/doi/full/10.15252/msb.20209942>Enabling high‚Äêthroughput biology with flexible open‚Äêsource automation</a></a></li><li><a href=#cytotree-an-rbioconductor-package-for-analysis-and-visualization-of-flow-and-mass-cytometry-datahttpsbmcbioinformaticsbiomedcentralcomarticles101186s12859-021-04054-2><a href=https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04054-2>CytoTree: an R/Bioconductor package for analysis and visualization of flow and mass cytometry data</a></a></li></ul></li><li><a href=#i-classfar-fa-keyboardi-programming><i class="far fa-keyboard"></i> Programming</a><ul><li><a href=#how-we-built-our-python-client-thats-mostly-rusthttpswwwfluvioioblog202103python-client><a href=https://www.fluvio.io/blog/2021/03/python-client/>How we built our Python Client that&rsquo;s mostly Rust</a></a></li><li><a href=#how-to-implement-a-hash-table-in-chttpsbenhoytcomwritingshash-table-in-c><a href=https://benhoyt.com/writings/hash-table-in-c/>How to implement a hash table (in C)</a></a></li><li><a href=#c-in-python-the-easy-wayhttpswwwyoutubecomwatchv_5t70caxdj0><a href="https://www.youtube.com/watch?v=_5T70cAXDJ0">C++ in Python the Easy Way!</a></a></li></ul></li><li><a href=#i-classfas-fa-toolboxi-tools><i class="fas fa-toolbox"></i> Tools</a><ul><li><a href=#vitals-a-tiny-macos-process-monitorhttpshmarrcomblogvitals><a href=https://hmarr.com/blog/vitals/>Vitals: a tiny macOS process monitor</a></a></li><li><a href=#wang0618pywebiohttpsgithubcomwang0618pywebio><a href=https://github.com/wang0618/PyWebIO>wang0618/PyWebIO</a></a></li><li><a href=#synekgit-planhttpsgithubcomsynekgit-plan><a href=https://github.com/synek/git-plan>synek/git-plan</a></a></li><li><a href=#textpatternhttpstextpatterncom><a href=https://textpattern.com/>Textpattern</a></a></li><li><a href=#create-your-own-vs-code-themeshttpsthemesvscodeone><a href=https://themes.vscode.one/>Create your own VS Code themes</a></a></li><li><a href=#hextedithttpshexteditapp><a href=https://hextedit.app/>HextEdit</a></a></li><li><a href=#grunsicecreamhttpsgithubcomgrunsicecream><a href=https://github.com/gruns/icecream>gruns/icecream</a></a></li><li><a href=#benfredpy-spyhttpsgithubcombenfredpy-spy><a href=https://github.com/benfred/py-spy>benfred/py-spy</a></a></li></ul></li><li><a href=#i-classfas-fa-graduation-capi-guides-and-tutorials><i class="fas fa-graduation-cap"></i> Guides and Tutorials</a><ul><li><a href=#my-python-testing-style-guidehttpsblogtheacodesmy-python-testing-style-guide><a href=https://blog.thea.codes/my-python-testing-style-guide/>My Python testing style guide</a></a></li><li><a href=#sqlite-is-not-a-toy-databasehttpsantonzorgsqlite-is-not-a-toy-database><a href=https://antonz.org/sqlite-is-not-a-toy-database/>SQLite is not a toy database</a></a></li><li><a href=#socathttpscopyconstructmediumcomsocat-29453e9fc8a6><a href=https://copyconstruct.medium.com/socat-29453e9fc8a6>socat</a></a></li><li><a href=#the-worst-so-called-best-practice-for-dockerhttpspythonspeedcomarticlessecurity-updates-in-docker><a href=https://pythonspeed.com/articles/security-updates-in-docker/>The worst so-called ‚Äúbest practice‚Äù for Docker</a></a></li><li><a href=#sds-375httpswilkelaborgsds375><a href=https://wilkelab.org/SDS375/>SDS 375</a></a></li></ul></li><li><a href=#i-classfas-fa-rssi-others><i class="fas fa-rss"></i> Others</a><ul><li><a href=#the-fastest-way-to-loop-in-python---an-unfortunate-truthhttpswwwyoutubecomwatchvqgevy75co8c><a href="https://www.youtube.com/watch?v=Qgevy75co8c">The Fastest Way to Loop in Python - An Unfortunate Truth</a></a></li><li><a href=#ssh-keygen-a-2021-updatehttpsblogrebasedpl20210323ssh-keygen-2021html><a href=https://blog.rebased.pl/2021/03/23/ssh-keygen-2021.html>SSH Keygen: a 2021 update</a></a></li><li><a href=#medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drivehttpswwwvicecomenarticle5dp7y3medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drive><a href=https://www.vice.com/en/article/5dp7y3/medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drive>Medium Tells Journalists to Feel Free to Quit After Busting Union Drive</a></a></li><li><a href=#fundamentals-of-optimal-code-stylehttpsoptimal-codestylegithubio><a href=https://optimal-codestyle.github.io/>Fundamentals of Optimal Code Style</a></a></li><li><a href=#building-a-full-text-search-engine-in-150-lines-of-python-codehttpsbartdegoedebuilding-a-full-text-search-engine-150-lines-of-code><a href=https://bart.degoe.de/building-a-full-text-search-engine-150-lines-of-code/>Building a full-text search engine in 150 lines of Python code</a></a></li></ul></li><li><a href=#i-classfas-fa-rssi-others-1><i class="fas fa-rss"></i> Others</a><ul><li><a href=#opencellhttpsopencellczbiohuborg><a href=https://opencell.czbiohub.org/>OpenCell</a></a></li><li><a href=#study-shows-survival-mechanism-for-cells-under-stresshttpswwwkthseenaktuelltnyheterstudy-shows-survival-mechanism-for-cells-under-stress-11061825><a href=https://www.kth.se/en/aktuellt/nyheter/study-shows-survival-mechanism-for-cells-under-stress-1.1061825>Study shows survival mechanism for cells under stress</a></a></li><li><a href=#this-is-perhaps-the-first-organism-whose-evolutionary-history-was-in-a-computerhttpsthebiologistrsborgukbiologist-featuresprofessor-michael-levin-interview><a href=https://thebiologist.rsb.org.uk/biologist-features/professor-michael-levin-interview>‚ÄòThis is perhaps the first organism whose evolutionary history was in a computer‚Äô</a></a></li></ul></li></ul></nav></div></details></div></div><div class="min-w-0 min-h-0 max-w-prose grow"><h2 id=i-classfas-fa-bullhorni-news class="relative group"><i class="fas fa-bullhorn"></i> News <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-bullhorni-news aria-label=Anchor>#</a></span></h2><ul><li><a href=https://news.itsfoss.com/richard-stallman-is-back-at-fsf/ target=_blank rel="noreferrer noopener">Richard Stallman is Back!</a></li><li><a href=https://forty.gnome.org/ target=_blank rel="noreferrer noopener">Gnome 40 is out with some major UI changes</a></li></ul><hr><h2 id=i-classfas-fa-dnai-sciencebioinformatics class="relative group"><i class="fas fa-dna"></i> Science/Bioinformatics <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-dnai-sciencebioinformatics aria-label=Anchor>#</a></span></h2><h3 id=abel-prize-celebrates-union-of-mathematics-and-computer-sciencehttpswwwnaturecomarticlesd41586-021-00694-9 class="relative group"><a href=https://www.nature.com/articles/d41586-021-00694-9 target=_blank rel="noreferrer noopener">Abel Prize celebrates union of mathematics and computer science</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#abel-prize-celebrates-union-of-mathematics-and-computer-sciencehttpswwwnaturecomarticlesd41586-021-00694-9 aria-label=Anchor>#</a></span></h3><p><em>Davide Castelvecchi</em></p><p><a href=# class="badge badge-primary">#abel-prize</a>
<a href=# class="badge badge-primary">#mathematics</a></p><p><em>Two pioneers of the theory of computation have won the 2021 Abel Prize, one of the most prestigious honours in mathematics.</em></p><h3 id=an-overview-of-next-generation-sequencinghttpswwwtechnologynetworkscomgenomicsarticlesan-overview-of-next-generation-sequencing-346532 class="relative group"><a href=https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532 target=_blank rel="noreferrer noopener">An Overview of Next-Generation Sequencing</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#an-overview-of-next-generation-sequencinghttpswwwtechnologynetworkscomgenomicsarticlesan-overview-of-next-generation-sequencing-346532 aria-label=Anchor>#</a></span></h3><p><em>Athina Gkazi, PhD</em></p><p><a href=# class="badge badge-primary">#ngs</a>
<a href=# class="badge badge-primary">#overview</a></p><p>Over the last 56 years, researchers have been developing methods and technologies to assist in the determination of nucleic acid sequences in biological samples. Our ability to sequence DNA and RNA accurately has had a great impact in numerous research fields. This article discusses what next-generation sequencing (NGS) is, advances in the technology and its applications.</p><h3 id=robust-integration-of-multiple-single-cell-rna-sequencing-datasets-using-a-single-reference-spacehttpswwwnaturecomarticless41587-021-00859-x class="relative group"><a href=https://www.nature.com/articles/s41587-021-00859-x target=_blank rel="noreferrer noopener">Robust integration of multiple single-cell RNA sequencing datasets using a single reference space</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#robust-integration-of-multiple-single-cell-rna-sequencing-datasets-using-a-single-reference-spacehttpswwwnaturecomarticless41587-021-00859-x aria-label=Anchor>#</a></span></h3><p><em>Liu et al., Nat Biotechnol (2021)</em></p><p><a href=# class="badge badge-primary">#scrna-seq</a>
<a href=# class="badge badge-primary">#integration</a></p><p><em>In many biological applications of single-cell RNA sequencing (scRNA-seq), an integrated analysis of data from multiple batches or studies is necessary. Current methods typically achieve integration using shared cell types or covariance correlation between datasets, which can distort biological signals. Here we introduce an algorithm that uses the gene eigenvectors from a reference dataset to establish a global frame for integration. Using simulated and real datasets, we demonstrate that this approach, called Reference Principal Component Integration (RPCI), consistently outperforms other methods by multiple metrics, with clear advantages in preserving genuine cross-sample gene expression differences in matching cell types, such as those present in cells at distinct developmental stages or in perturbated versus control studies. Moreover, RPCI maintains this robust performance when multiple datasets are integrated. Finally, we applied RPCI to scRNA-seq data for mouse gut endoderm development and revealed temporal emergence of genetic programs helping establish the anterior‚Äìposterior axis in visceral endoderm.</em></p><h3 id=integrated-intra-and-intercellular-signaling-knowledge-for-multicellular-omics-analysishttpswwwembopressorgdoifull1015252msb20209923 class="relative group"><a href=https://www.embopress.org/doi/full/10.15252/msb.20209923 target=_blank rel="noreferrer noopener">Integrated intra‚Äê and intercellular signaling knowledge for multicellular omics analysis</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#integrated-intra-and-intercellular-signaling-knowledge-for-multicellular-omics-analysishttpswwwembopressorgdoifull1015252msb20209923 aria-label=Anchor>#</a></span></h3><p><em>Turei et al., Mol Sys Biol (2021)</em></p><p><a href=# class="badge badge-primary">#signaling</a>
<a href=# class="badge badge-primary">#cell-interactions</a></p><p><em>Molecular knowledge of biological processes is a cornerstone in omics data analysis. Applied to single‚Äêcell data, such analyses provide mechanistic insights into individual cells and their interactions. However, knowledge of intercellular communication is scarce, scattered across resources, and not linked to intracellular processes. To address this gap, we combined over 100 resources covering interactions and roles of proteins in inter‚Äê and intracellular signaling, as well as transcriptional and post‚Äêtranscriptional regulation. We added protein complex information and annotations on function, localization, and role in diseases for each protein. The resource is available for human, and via homology translation for mouse and rat. The data are accessible via OmniPath‚Äôs web service (<a href=https://omnipathdb.org/%29 target=_blank rel="noreferrer noopener">https://omnipathdb.org/)</a>, a Cytoscape plug‚Äêin, and packages in R/Bioconductor and Python, providing access options for computational and experimental scientists. We created workflows with tutorials to facilitate the analysis of cell‚Äìcell interactions and affected downstream intracellular signaling processes. OmniPath provides a single access point to knowledge spanning intra‚Äê and intercellular processes for data analysis, as we demonstrate in applications studying SARS‚ÄêCoV‚Äê2 infection and ulcerative colitis.</em></p><h3 id=sepal-identifying-transcript-profiles-with-spatial-patterns-by-diffusion-based-modelinghttpsacademicoupcombioinformaticsadvance-articledoi101093bioinformaticsbtab1646168120 class="relative group"><a href=https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btab164/6168120 target=_blank rel="noreferrer noopener">sepal: identifying transcript profiles with spatial patterns by diffusion-based modeling</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#sepal-identifying-transcript-profiles-with-spatial-patterns-by-diffusion-based-modelinghttpsacademicoupcombioinformaticsadvance-articledoi101093bioinformaticsbtab1646168120 aria-label=Anchor>#</a></span></h3><p><em>Andersson et al., Bioinformatics (2021)</em></p><p><a href=# class="badge badge-primary">#spatial-transcriptomics</a></p><p><em>Collection of spatial signals in large numbers has become a routine task in multiple omics-fields, but parsing of these rich datasets still pose certain challenges. In whole or near-full transcriptome spatial techniques, spurious expression profiles are intermixed with those exhibiting an organized structure. To distinguish profiles with spatial patterns from the background noise, a metric that enables quantification of spatial structure is desirable. Current methods designed for similar purposes tend to be built around a framework of statistical hypothesis testing, hence we were compelled to explore a fundamentally different strategy. We propose an unexplored approach to analyze spatial transcriptomics data, simulating diffusion of individual transcripts to extract genes with spatial patterns. The method performed as expected when presented with synthetic data. When applied to real data, it identified genes with distinct spatial profiles, involved in key biological processes or characteristic for certain cell types. Compared to existing methods, ours seemed to be less informed by the genes‚Äô expression levels and showed better time performance when run with multiple cores.</em></p><h3 id=user-friendly-scalable-tools-and-workflows-for-single-cell-rna-seq-analysishttpswwwnaturecomarticless41592-021-01102-w class="relative group"><a href=https://www.nature.com/articles/s41592-021-01102-w target=_blank rel="noreferrer noopener">User-friendly, scalable tools and workflows for single-cell RNA-seq analysis</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#user-friendly-scalable-tools-and-workflows-for-single-cell-rna-seq-analysishttpswwwnaturecomarticless41592-021-01102-w aria-label=Anchor>#</a></span></h3><p><em>Moreno et al., Nat Methods (2021)</em></p><p><a href=# class="badge badge-primary">#portal</a>
<a href=# class="badge badge-primary">#scrna-seq</a></p><p><em>As single-cell RNA sequencing (scRNA-seq) becomes widespread, accessible and scalable computational pipelines for data analysis are needed. We introduce an interactive computational environment for single-cell studies based on Galaxy1, with functions from established workflows. Single Cell Interactive Application (SCiAp) provides easy access to data from the Human Cell Atlas (HCA) and EMBL-EBI‚Äôs Single Cell Expression Atlas (SCEA)2 projects and can be deployed on different computing platforms, making single-cell data analysis of large-scale projects accessible to the scientific community.</em></p><h3 id=enabling-highthroughput-biology-with-flexible-opensource-automationhttpswwwembopressorgdoifull1015252msb20209942 class="relative group"><a href=https://www.embopress.org/doi/full/10.15252/msb.20209942 target=_blank rel="noreferrer noopener">Enabling high‚Äêthroughput biology with flexible open‚Äêsource automation</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#enabling-highthroughput-biology-with-flexible-opensource-automationhttpswwwembopressorgdoifull1015252msb20209942 aria-label=Anchor>#</a></span></h3><p><em>Chory et al., Mol Sys Biol (2021)</em></p><p><a href=# class="badge badge-primary">#automation</a>
<a href=# class="badge badge-primary">#lab</a></p><p><em>Our understanding of complex living systems is limited by our capacity to perform experiments in high throughput. While robotic systems have automated many traditional hand‚Äêpipetting protocols, software limitations have precluded more advanced maneuvers required to manipulate, maintain, and monitor hundreds of experiments in parallel. Here, we present Pyhamilton, an open‚Äêsource Python platform that can execute complex pipetting patterns required for custom high‚Äêthroughput experiments such as the simulation of metapopulation dynamics. With an integrated plate reader, we maintain nearly 500 remotely monitored bacterial cultures in log‚Äêphase growth for days without user intervention by taking regular density measurements to adjust the robotic method in real‚Äêtime. Using these capabilities, we systematically optimize bioreactor protein production by monitoring the fluorescent protein expression and growth rates of a hundred different continuous culture conditions in triplicate to comprehensively sample the carbon, nitrogen, and phosphorus fitness landscape. Our results demonstrate that flexible software can empower existing hardware to enable new types and scales of experiments, empowering areas from biomanufacturing to fundamental biology.</em></p><h3 id=cytotree-an-rbioconductor-package-for-analysis-and-visualization-of-flow-and-mass-cytometry-datahttpsbmcbioinformaticsbiomedcentralcomarticles101186s12859-021-04054-2 class="relative group"><a href=https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04054-2 target=_blank rel="noreferrer noopener">CytoTree: an R/Bioconductor package for analysis and visualization of flow and mass cytometry data</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#cytotree-an-rbioconductor-package-for-analysis-and-visualization-of-flow-and-mass-cytometry-datahttpsbmcbioinformaticsbiomedcentralcomarticles101186s12859-021-04054-2 aria-label=Anchor>#</a></span></h3><p><em>Dai et al., BMC Bioinformatics (2021)</em></p><p><a href=# class="badge badge-primary">#cytometry</a></p><p><em>The rapidly increasing dimensionality and throughput of flow and mass cytometry data necessitate new bioinformatics tools for analysis and interpretation, and the recently emerging single-cell-based algorithms provide a powerful strategy to meet this challenge. Here, we present CytoTree, an R/Bioconductor package designed to analyze and interpret multidimensional flow and mass cytometry data. CytoTree provides multiple computational functionalities that integrate most of the commonly used techniques in unsupervised clustering and dimensionality reduction and, more importantly, support the construction of a tree-shaped trajectory based on the minimum spanning tree algorithm. A graph-based algorithm is also implemented to estimate the pseudotime and infer intermediate-state cells. We apply CytoTree to several examples of mass cytometry and time-course flow cytometry data on heterogeneity-based cytology and differentiation/reprogramming experiments to illustrate the practical utility achieved in a fast and convenient manner. CytoTree represents a versatile tool for analyzing multidimensional flow and mass cytometry data and to producing heuristic results for trajectory construction and pseudotime estimation in an integrated workflow.</em></p><hr><h2 id=i-classfar-fa-keyboardi-programming class="relative group"><i class="far fa-keyboard"></i> Programming <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfar-fa-keyboardi-programming aria-label=Anchor>#</a></span></h2><h3 id=how-we-built-our-python-client-thats-mostly-rusthttpswwwfluvioioblog202103python-client class="relative group"><a href=https://www.fluvio.io/blog/2021/03/python-client/ target=_blank rel="noreferrer noopener">How we built our Python Client that&rsquo;s mostly Rust</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#how-we-built-our-python-client-thats-mostly-rusthttpswwwfluvioioblog202103python-client aria-label=Anchor>#</a></span></h3><p>In this post, we‚Äôll talk about how we were able to leverage some of the great Rust tools to build a Python client without writing much Python itself.</p><h3 id=how-to-implement-a-hash-table-in-chttpsbenhoytcomwritingshash-table-in-c class="relative group"><a href=https://benhoyt.com/writings/hash-table-in-c/ target=_blank rel="noreferrer noopener">How to implement a hash table (in C)</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#how-to-implement-a-hash-table-in-chttpsbenhoytcomwritingshash-table-in-c aria-label=Anchor>#</a></span></h3><p>Summary: An explanation of how to implement a simple hash table data structure using the C programming language.
I briefly demonstrate linear and binary search, and then design and implement a hash table.
My goal is to show that hash table internals are not scary, but ‚Äì within certain constraints ‚Äì are easy enough to build from scratch.</p><h3 id=c-in-python-the-easy-wayhttpswwwyoutubecomwatchv_5t70caxdj0 class="relative group"><a href="https://www.youtube.com/watch?v=_5T70cAXDJ0" target=_blank rel="noreferrer noopener">C++ in Python the Easy Way!</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#c-in-python-the-easy-wayhttpswwwyoutubecomwatchv_5t70caxdj0 aria-label=Anchor>#</a></span></h3><p>Youtube tutorial on how to wrap C++ in python using pybind.</p><hr><h2 id=i-classfas-fa-toolboxi-tools class="relative group"><i class="fas fa-toolbox"></i> Tools <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-toolboxi-tools aria-label=Anchor>#</a></span></h2><h3 id=vitals-a-tiny-macos-process-monitorhttpshmarrcomblogvitals class="relative group"><a href=https://hmarr.com/blog/vitals/ target=_blank rel="noreferrer noopener">Vitals: a tiny macOS process monitor</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#vitals-a-tiny-macos-process-monitorhttpshmarrcomblogvitals aria-label=Anchor>#</a></span></h3><p>Vitals is a little Mac app I built recently that shows you which programs are slowing your computer down. It lives in the menu bar, keeping track of resource usage in the background, so you can summon it instantly at the click of a button.</p><h3 id=wang0618pywebiohttpsgithubcomwang0618pywebio class="relative group"><a href=https://github.com/wang0618/PyWebIO target=_blank rel="noreferrer noopener">wang0618/PyWebIO</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#wang0618pywebiohttpsgithubcomwang0618pywebio aria-label=Anchor>#</a></span></h3><p>Write interactive web app in script way.</p><h3 id=synekgit-planhttpsgithubcomsynekgit-plan class="relative group"><a href=https://github.com/synek/git-plan target=_blank rel="noreferrer noopener">synek/git-plan</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#synekgit-planhttpsgithubcomsynekgit-plan aria-label=Anchor>#</a></span></h3><p>Git Plan - a better workflow for git</p><h3 id=textpatternhttpstextpatterncom class="relative group"><a href=https://textpattern.com/ target=_blank rel="noreferrer noopener">Textpattern</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#textpatternhttpstextpatterncom aria-label=Anchor>#</a></span></h3><p>The small content management system that can handle big ideas</p><h3 id=create-your-own-vs-code-themeshttpsthemesvscodeone class="relative group"><a href=https://themes.vscode.one/ target=_blank rel="noreferrer noopener">Create your own VS Code themes</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#create-your-own-vs-code-themeshttpsthemesvscodeone aria-label=Anchor>#</a></span></h3><p>Hey! My name is Mike, and I built the VS Code theme studio so you could easily design, deploy, and share your own VS Code themes from scratch. Enjoy!</p><h3 id=hextedithttpshexteditapp class="relative group"><a href=https://hextedit.app/ target=_blank rel="noreferrer noopener">HextEdit</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#hextedithttpshexteditapp aria-label=Anchor>#</a></span></h3><p>A fast and native hex editor for macOS</p><h3 id=grunsicecreamhttpsgithubcomgrunsicecream class="relative group"><a href=https://github.com/gruns/icecream target=_blank rel="noreferrer noopener">gruns/icecream</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#grunsicecreamhttpsgithubcomgrunsicecream aria-label=Anchor>#</a></span></h3><p>üç¶ Never use print() to debug again.</p><h3 id=benfredpy-spyhttpsgithubcombenfredpy-spy class="relative group"><a href=https://github.com/benfred/py-spy target=_blank rel="noreferrer noopener">benfred/py-spy</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#benfredpy-spyhttpsgithubcombenfredpy-spy aria-label=Anchor>#</a></span></h3><p>Sampling profiler for Python programs</p><hr><h2 id=i-classfas-fa-graduation-capi-guides-and-tutorials class="relative group"><i class="fas fa-graduation-cap"></i> Guides and Tutorials <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-graduation-capi-guides-and-tutorials aria-label=Anchor>#</a></span></h2><h3 id=my-python-testing-style-guidehttpsblogtheacodesmy-python-testing-style-guide class="relative group"><a href=https://blog.thea.codes/my-python-testing-style-guide/ target=_blank rel="noreferrer noopener">My Python testing style guide</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#my-python-testing-style-guidehttpsblogtheacodesmy-python-testing-style-guide aria-label=Anchor>#</a></span></h3><p>Stargirl Flowers&rsquo; python practices for testing.</p><h3 id=sqlite-is-not-a-toy-databasehttpsantonzorgsqlite-is-not-a-toy-database class="relative group"><a href=https://antonz.org/sqlite-is-not-a-toy-database/ target=_blank rel="noreferrer noopener">SQLite is not a toy database</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#sqlite-is-not-a-toy-databasehttpsantonzorgsqlite-is-not-a-toy-database aria-label=Anchor>#</a></span></h3><p>Whether you are a developer, data analyst, QA engineer, DevOps person, or product manager - SQLite is a perfect tool for you. Here is why.</p><h3 id=socathttpscopyconstructmediumcomsocat-29453e9fc8a6 class="relative group"><a href=https://copyconstruct.medium.com/socat-29453e9fc8a6 target=_blank rel="noreferrer noopener">socat</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#socathttpscopyconstructmediumcomsocat-29453e9fc8a6 aria-label=Anchor>#</a></span></h3><p>socat stands for SOcket CAT. It is a utility for data transfer between two addresses.</p><h3 id=the-worst-so-called-best-practice-for-dockerhttpspythonspeedcomarticlessecurity-updates-in-docker class="relative group"><a href=https://pythonspeed.com/articles/security-updates-in-docker/ target=_blank rel="noreferrer noopener">The worst so-called ‚Äúbest practice‚Äù for Docker</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#the-worst-so-called-best-practice-for-dockerhttpspythonspeedcomarticlessecurity-updates-in-docker aria-label=Anchor>#</a></span></h3><p>Think about switching to non-root in your container.</p><h3 id=sds-375httpswilkelaborgsds375 class="relative group"><a href=https://wilkelab.org/SDS375/ target=_blank rel="noreferrer noopener">SDS 375</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#sds-375httpswilkelaborgsds375 aria-label=Anchor>#</a></span></h3><p>Data Visualization in R</p><hr><h2 id=i-classfas-fa-rssi-others class="relative group"><i class="fas fa-rss"></i> Others <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-rssi-others aria-label=Anchor>#</a></span></h2><h3 id=the-fastest-way-to-loop-in-python---an-unfortunate-truthhttpswwwyoutubecomwatchvqgevy75co8c class="relative group"><a href="https://www.youtube.com/watch?v=Qgevy75co8c" target=_blank rel="noreferrer noopener">The Fastest Way to Loop in Python - An Unfortunate Truth</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#the-fastest-way-to-loop-in-python---an-unfortunate-truthhttpswwwyoutubecomwatchvqgevy75co8c aria-label=Anchor>#</a></span></h3><p>What is faster, a for loop, a while loop, or something else? In this video, we try several different ways to accomplish a looping task and discover which is fastest.</p><h3 id=ssh-keygen-a-2021-updatehttpsblogrebasedpl20210323ssh-keygen-2021html class="relative group"><a href=https://blog.rebased.pl/2021/03/23/ssh-keygen-2021.html target=_blank rel="noreferrer noopener">SSH Keygen: a 2021 update</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#ssh-keygen-a-2021-updatehttpsblogrebasedpl20210323ssh-keygen-2021html aria-label=Anchor>#</a></span></h3><p>Read more about new feature of signing your SSH key.</p><h3 id=medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drivehttpswwwvicecomenarticle5dp7y3medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drive class="relative group"><a href=https://www.vice.com/en/article/5dp7y3/medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drive target=_blank rel="noreferrer noopener">Medium Tells Journalists to Feel Free to Quit After Busting Union Drive</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drivehttpswwwvicecomenarticle5dp7y3medium-tells-journalists-to-feel-free-to-quit-after-busting-union-drive aria-label=Anchor>#</a></span></h3><p>After what workers describe as a successful union-busting campaign, Ev Williams has announced to journalists who work for him that they should feel free to go.</p><h3 id=fundamentals-of-optimal-code-stylehttpsoptimal-codestylegithubio class="relative group"><a href=https://optimal-codestyle.github.io/ target=_blank rel="noreferrer noopener">Fundamentals of Optimal Code Style</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#fundamentals-of-optimal-code-stylehttpsoptimal-codestylegithubio aria-label=Anchor>#</a></span></h3><p>Cognitively-Oriented Approach to Improving Program Readability by <em>Aleksandr Skobelev</em></p><h3 id=building-a-full-text-search-engine-in-150-lines-of-python-codehttpsbartdegoedebuilding-a-full-text-search-engine-150-lines-of-code class="relative group"><a href=https://bart.degoe.de/building-a-full-text-search-engine-150-lines-of-code/ target=_blank rel="noreferrer noopener">Building a full-text search engine in 150 lines of Python code</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#building-a-full-text-search-engine-in-150-lines-of-python-codehttpsbartdegoedebuilding-a-full-text-search-engine-150-lines-of-code aria-label=Anchor>#</a></span></h3><p><strong>Personal recommendation</strong></p><hr><h2 id=i-classfas-fa-rssi-others-1 class="relative group"><i class="fas fa-rss"></i> Others <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#i-classfas-fa-rssi-others-1 aria-label=Anchor>#</a></span></h2><h3 id=opencellhttpsopencellczbiohuborg class="relative group"><a href=https://opencell.czbiohub.org/ target=_blank rel="noreferrer noopener">OpenCell</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#opencellhttpsopencellczbiohuborg aria-label=Anchor>#</a></span></h3><p>Proteome-scale measurements of human protein localization and interactions</p><h3 id=study-shows-survival-mechanism-for-cells-under-stresshttpswwwkthseenaktuelltnyheterstudy-shows-survival-mechanism-for-cells-under-stress-11061825 class="relative group"><a href=https://www.kth.se/en/aktuellt/nyheter/study-shows-survival-mechanism-for-cells-under-stress-1.1061825 target=_blank rel="noreferrer noopener">Study shows survival mechanism for cells under stress</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#study-shows-survival-mechanism-for-cells-under-stresshttpswwwkthseenaktuelltnyheterstudy-shows-survival-mechanism-for-cells-under-stress-11061825 aria-label=Anchor>#</a></span></h3><p>New research reveals how cancer cells endure stress and survive. Publishing in Molecular Cell, an international research team identified mechanisms that human and mouse cells use to survive heat shock and resume their original function - and even pass the memory of the experience of stress down to their daughter cells.</p><h3 id=this-is-perhaps-the-first-organism-whose-evolutionary-history-was-in-a-computerhttpsthebiologistrsborgukbiologist-featuresprofessor-michael-levin-interview class="relative group"><a href=https://thebiologist.rsb.org.uk/biologist-features/professor-michael-levin-interview target=_blank rel="noreferrer noopener">‚ÄòThis is perhaps the first organism whose evolutionary history was in a computer‚Äô</a> <span class="absolute top-0 w-6 transition-opacity opacity-0 -start-6 not-prose group-hover:opacity-100"><a class="group-hover:text-primary-300 dark:group-hover:text-neutral-700" style=text-decoration-line:none!important href=#this-is-perhaps-the-first-organism-whose-evolutionary-history-was-in-a-computerhttpsthebiologistrsborgukbiologist-featuresprofessor-michael-levin-interview aria-label=Anchor>#</a></span></h3><p>Michael Levin‚Äôs work explores how individual cells co-ordinate into multicellular shapes. As Tom Ireland finds, this has not only led to the creation of ‚Äòliving robots‚Äô made entirely from frog cells, but it could be used to repair birth defects, regenerate damaged tissue and reprogramme tumours.</p></div></section><footer class="pt-8 max-w-prose print:hidden"><div class=flex><img class="!mb-0 !mt-0 me-4 h-24 w-24 rounded-full" width=96 height=96 alt="Martin Proks" src=/img/me_hu62f706925fb4739dcc3fb379f3b73c90_43704_192x192_fill_box_center_3.png loading=lazy><div class=place-self-center><div class="text-[0.6rem] uppercase leading-3 text-neutral-500 dark:text-neutral-400">Author</div><div class="font-semibold leading-6 text-neutral-800 dark:text-neutral-300">Martin Proks</div><div class="text-sm text-neutral-700 dark:text-neutral-400">I am a Bioinformatics PhD student @reNEW, University of Copenhagen in Denmark. My primary focus is Computational stem cell research.</div><div class="text-2xl sm:text-lg"><div class="flex flex-wrap text-neutral-400 dark:text-neutral-500"><a class="px-1 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400" style=will-change:transform href=https://twitter.com/mproksik target=_blank aria-label=Twitter rel="me noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M459.37 151.716c.325 4.548.325 9.097.325 13.645.0 138.72-105.583 298.558-298.558 298.558-59.452.0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055.0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421.0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391.0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04.0-57.828 46.782-104.934 104.934-104.934 30.213.0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg></span></a><a class="px-1 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400" style=will-change:transform href=https://linkedin.com/in/martinproks target=_blank aria-label=Linkedin rel="me noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentcolor" d="M416 32H31.9C14.3 32 0 46.5.0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6.0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3.0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2.0 38.5 17.3 38.5 38.5.0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6.0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2.0 79.7 44.3 79.7 101.9V416z"/></svg></span></a><a class="px-1 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400" style=will-change:transform href=https://github.com/matq007 target=_blank aria-label=Github rel="me noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><path fill="currentcolor" d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6.0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6.0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3.0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1.0-6.2-.3-40.4-.3-61.4.0.0-70 15-84.7-29.8.0.0-11.4-29.1-27.8-36.6.0.0-22.9-15.7 1.6-15.4.0.0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5.0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9.0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4.0 33.7-.3 75.4-.3 83.6.0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6.0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9.0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg></span></a><a class="px-1 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400" style=will-change:transform href=https://orcid.org/0000-0002-8178-3128 target=_blank aria-label=Orcid rel="me noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M294.75 188.19h-45.92V342h47.47c67.62.0 83.12-51.34 83.12-76.91.0-41.64-26.54-76.9-84.67-76.9zM256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm-80.79 360.76h-29.84v-207.5h29.84zm-14.92-231.14a19.57 19.57.0 1119.57-19.57 19.64 19.64.0 01-19.57 19.57zM3e2 369h-81V161.26h80.6c76.73.0 110.44 54.83 110.44 103.85C410 318.39 368.38 369 3e2 369z"/></svg></span></a><a class="px-1 transition-transform hover:scale-125 hover:text-primary-700 dark:hover:text-primary-400" style=will-change:transform href=https://www.researchgate.net/profile/Martin-Proks target=_blank aria-label=Researchgate rel="me noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentcolor" d="M0 32v448h448V32H0zm262.2 334.4c-6.6 3-33.2 6-50-14.2-9.2-10.6-25.3-33.3-42.2-63.6-8.9.0-14.7.0-21.4-.6v46.4c0 23.5 6 21.2 25.8 23.9v8.1c-6.9-.3-23.1-.8-35.6-.8-13.1.0-26.1.6-33.6.8v-8.1c15.5-2.9 22-1.3 22-23.9V225c0-22.6-6.4-21-22-23.9V193c25.8 1 53.1-.6 70.9-.6 31.7.0 55.9 14.4 55.9 45.6.0 21.1-16.7 42.2-39.2 47.5 13.6 24.2 30 45.6 42.2 58.9 7.2 7.8 17.2 14.7 27.2 14.7v7.3zm22.9-135c-23.3.0-32.2-15.7-32.2-32.2V167c0-12.2 8.8-30.4 34-30.4s30.4 17.9 30.4 17.9l-10.7 7.2s-5.5-12.5-19.7-12.5c-7.9.0-19.7 7.3-19.7 19.7v26.8c0 13.4 6.6 23.3 17.9 23.3 14.1.0 21.5-10.9 21.5-26.8h-17.9v-10.7h30.4c0 20.5 4.7 49.9-34 49.9zm-116.5 44.7c-9.4.0-13.6-.3-20-.8v-69.7c6.4-.6 15-.6 22.5-.6 23.3.0 37.2 12.2 37.2 34.5.0 21.9-15 36.6-39.7 36.6z"/></svg></span></a></div></div></div></div><section class="flex flex-row flex-wrap justify-center pt-4 text-xl"><a class="m-1 inline-block min-w-[2.4rem] rounded bg-neutral-300 p-1 text-center text-neutral-700 hover:bg-primary-500 hover:text-neutral dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800" href="https://www.facebook.com/sharer/sharer.php?u=/weekly/weekly-news-13-14/&amp;quote=Bi-Weekly%20news%20#13%20and%20#14" title="Share on Facebook" aria-label="Share on Facebook" target=_blank rel="noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14.0 55.52 4.84 55.52 4.84v61h-31.28c-30.8.0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"/></svg></span></a><a class="m-1 inline-block min-w-[2.4rem] rounded bg-neutral-300 p-1 text-center text-neutral-700 hover:bg-primary-500 hover:text-neutral dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800" href="https://twitter.com/intent/tweet/?url=/weekly/weekly-news-13-14/&amp;text=Bi-Weekly%20news%20#13%20and%20#14" title="Tweet on Twitter" aria-label="Tweet on Twitter" target=_blank rel="noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M459.37 151.716c.325 4.548.325 9.097.325 13.645.0 138.72-105.583 298.558-298.558 298.558-59.452.0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055.0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421.0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391.0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04.0-57.828 46.782-104.934 104.934-104.934 30.213.0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"/></svg></span></a><a class="m-1 inline-block min-w-[2.4rem] rounded bg-neutral-300 p-1 text-center text-neutral-700 hover:bg-primary-500 hover:text-neutral dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800" href="https://tootpick.org/#text=/weekly/weekly-news-13-14/%20Bi-Weekly%20news%20#13%20and%20#14" title="Toot on Mastodon" aria-label="Toot on Mastodon" target=_blank rel="noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentcolor" d="M433 179.11c0-97.2-63.71-125.7-63.71-125.7-62.52-28.7-228.56-28.4-290.48.0.0.0-63.72 28.5-63.72 125.7.0 115.7-6.6 259.4 105.63 289.1 40.51 10.7 75.32 13 103.33 11.4 50.81-2.8 79.32-18.1 79.32-18.1l-1.7-36.9s-36.31 11.4-77.12 10.1c-40.41-1.4-83-4.4-89.63-54a102.54 102.54.0 01-.9-13.9c85.63 20.9 158.65 9.1 178.75 6.7 56.12-6.7 105-41.3 111.23-72.9 9.8-49.8 9-121.5 9-121.5zm-75.12 125.2h-46.63v-114.2c0-49.7-64-51.6-64 6.9v62.5h-46.33V197c0-58.5-64-56.6-64-6.9v114.2H90.19c0-122.1-5.2-147.9 18.41-175 25.9-28.9 79.82-30.8 103.83 6.1l11.6 19.5 11.6-19.5c24.11-37.1 78.12-34.8 103.83-6.1 23.71 27.3 18.4 53 18.4 175z"/></svg></span></a><a class="m-1 inline-block min-w-[2.4rem] rounded bg-neutral-300 p-1 text-center text-neutral-700 hover:bg-primary-500 hover:text-neutral dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800" href="https://reddit.com/submit/?url=/weekly/weekly-news-13-14/&amp;resubmit=true&amp;title=Bi-Weekly%20news%20#13%20and%20#14" title="Submit to Reddit" aria-label="Submit to Reddit" target=_blank rel="noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M201.5 305.5c-13.8.0-24.9-11.1-24.9-24.6.0-13.8 11.1-24.9 24.9-24.9 13.6.0 24.6 11.1 24.6 24.9.0 13.6-11.1 24.6-24.6 24.6zM504 256c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-132.3-41.2c-9.4.0-17.7 3.9-23.8 10-22.4-15.5-52.6-25.5-86.1-26.6l17.4-78.3 55.4 12.5c0 13.6 11.1 24.6 24.6 24.6 13.8.0 24.9-11.3 24.9-24.9s-11.1-24.9-24.9-24.9c-9.7.0-18 5.8-22.1 13.8l-61.2-13.6c-3-.8-6.1 1.4-6.9 4.4l-19.1 86.4c-33.2 1.4-63.1 11.3-85.5 26.8-6.1-6.4-14.7-10.2-24.1-10.2-34.9.0-46.3 46.9-14.4 62.8-1.1 5-1.7 10.2-1.7 15.5.0 52.6 59.2 95.2 132 95.2 73.1.0 132.3-42.6 132.3-95.2.0-5.3-.6-10.8-1.9-15.8 31.3-16 19.8-62.5-14.9-62.5zM302.8 331c-18.2 18.2-76.1 17.9-93.6.0-2.2-2.2-6.1-2.2-8.3.0-2.5 2.5-2.5 6.4.0 8.6 22.8 22.8 87.3 22.8 110.2.0 2.5-2.2 2.5-6.1.0-8.6-2.2-2.2-6.1-2.2-8.3.0zm7.7-75c-13.6.0-24.6 11.1-24.6 24.9.0 13.6 11.1 24.6 24.6 24.6 13.8.0 24.9-11.1 24.9-24.6.0-13.8-11-24.9-24.9-24.9z"/></svg></span></a><a class="m-1 inline-block min-w-[2.4rem] rounded bg-neutral-300 p-1 text-center text-neutral-700 hover:bg-primary-500 hover:text-neutral dark:bg-neutral-700 dark:text-neutral-300 dark:hover:bg-primary-400 dark:hover:text-neutral-800" href="https://www.linkedin.com/shareArticle?mini=true&amp;url=/weekly/weekly-news-13-14/&amp;title=Bi-Weekly%20news%20#13%20and%20#14" title="Share on LinkedIn" aria-label="Share on LinkedIn" target=_blank rel="noopener noreferrer"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentcolor" d="M416 32H31.9C14.3 32 0 46.5.0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6.0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3.0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2.0 38.5 17.3 38.5 38.5.0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6.0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2.0 79.7 44.3 79.7 101.9V416z"/></svg></span></a></section><div class=pt-8><hr class="border-dotted border-neutral-300 dark:border-neutral-600"><div class="flex justify-between pt-3"><span><a class="group flex" href=/weekly/weekly-news-12/><span class="me-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"><span class="ltr:inline rtl:hidden">&larr;</span><span class="ltr:hidden rtl:inline">&rarr;</span></span>
<span class="flex flex-col"><span class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500">Weekly news #12</span>
<span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400"><time datetime="2021-03-21 00:00:00 +0000 UTC">21 March 2021</time></span></span></a></span>
<span><a class="group flex text-right" href=/weekly/weekly-news-15/><span class="flex flex-col"><span class="mt-[0.1rem] leading-6 group-hover:underline group-hover:decoration-primary-500">Weekly news #15</span>
<span class="mt-[0.1rem] text-xs text-neutral-500 dark:text-neutral-400"><time datetime="2021-04-11 00:00:00 +0000 UTC">11 April 2021</time></span></span>
<span class="ms-2 text-neutral-700 transition-transform group-hover:-translate-x-[2px] group-hover:text-primary-600 dark:text-neutral dark:group-hover:text-primary-400"><span class="ltr:inline rtl:hidden">&rarr;</span><span class="ltr:hidden rtl:inline">&larr;</span></span></a></span></div></div></footer></article><div class="pointer-events-none absolute bottom-0 end-0 top-[100vh] w-12"><a href=#the-top class="pointer-events-auto sticky top-[calc(100vh-5.5rem)] flex h-12 w-12 items-center justify-center rounded-full bg-neutral/50 text-xl text-neutral-700 backdrop-blur hover:text-primary-600 dark:bg-neutral-800/50 dark:text-neutral dark:hover:text-primary-400" aria-label="Scroll to top" title="Scroll to top">&uarr;</a></div></main><footer class="py-10 print:hidden"><nav class="pb-4 text-base font-medium text-neutral-500 dark:text-neutral-400"><ul class="flex list-none flex-col sm:flex-row"><li class="group mb-1 ltr:text-right rtl:text-left sm:mb-0 sm:me-7 sm:last:me-0"><a href=/categories/ title=Categories><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Categories</span></a></li><li class="group mb-1 ltr:text-right rtl:text-left sm:mb-0 sm:me-7 sm:last:me-0"><a href=/tags/ title=Tags><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Tags</span></a></li><li class="group mb-1 ltr:text-right rtl:text-left sm:mb-0 sm:me-7 sm:last:me-0"><a href title><span class="decoration-primary-500 group-hover:underline group-hover:decoration-2 group-hover:underline-offset-2">Privacy</span></a></li></ul></nav><div class="flex items-center justify-between"><div><p class="text-sm text-neutral-500 dark:text-neutral-400">&copy;
2023
Martin Proks</p><p class="text-xs text-neutral-500 dark:text-neutral-400">Powered by <a class="hover:underline hover:decoration-primary-400 hover:text-primary-500" href=https://gohugo.io/ target=_blank rel="noopener noreferrer">Hugo</a> & <a class="hover:underline hover:decoration-primary-400 hover:text-primary-500" href=https://git.io/hugo-congo target=_blank rel="noopener noreferrer">Congo</a></p></div><div class="me-14 cursor-pointer text-sm text-neutral-700 hover:text-primary-600 dark:text-neutral dark:hover:text-primary-400"><button id=appearance-switcher-0 type=button aria-label="appearance switcher"><div class="flex h-12 w-12 items-center justify-center dark:hidden" title="Switch to dark appearance"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M32 256C32 132.2 132.3 32 255.8 32c11.36.0 29.7 1.668 40.9 3.746 9.616 1.777 11.75 14.63 3.279 19.44C245 86.5 211.2 144.6 211.2 207.8c0 109.7 99.71 193 208.3 172.3 9.561-1.805 16.28 9.324 10.11 16.95C387.9 448.6 324.8 480 255.8 480 132.1 480 32 379.6 32 256z"/></svg></span></div><div class="hidden h-12 w-12 items-center justify-center dark:flex" title="Switch to light appearance"><span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M256 159.1c-53.02.0-95.1 42.98-95.1 95.1s41.2 96.9 95.1 96.9 95.1-42.98 95.1-95.1S309 159.1 256 159.1zM509.3 347l-63.2-91.9 63.15-91.01c6.332-9.125 1.104-21.74-9.826-23.72l-109-19.7-19.7-109c-1.975-10.93-14.59-16.16-23.72-9.824L256 65.89 164.1 2.736c-9.125-6.332-21.74-1.107-23.72 9.824L121.6 121.6 12.56 141.3C1.633 143.2-3.596 155.9 2.736 164.1L65.89 256 2.74 347.01c-6.332 9.125-1.105 21.74 9.824 23.72l109 19.7 19.7 109c1.975 10.93 14.59 16.16 23.72 9.824L256 446.1l91.01 63.15c9.127 6.334 21.75 1.107 23.72-9.822l19.7-109 109-19.7C510.4 368.8 515.6 356.1 509.3 347zM256 383.1c-70.69.0-127.1-57.31-127.1-127.1.0-70.69 57.31-127.1 127.1-127.1S383.1 186.2 383.1 256c0 70.7-56.4 127.1-127.1 127.1z"/></svg></span></div></button></div></div></footer><div id=search-wrapper class="invisible fixed inset-0 z-50 flex h-screen w-screen cursor-default flex-col bg-neutral-500/50 p-4 backdrop-blur-sm dark:bg-neutral-900/50 sm:p-6 md:p-[10vh] lg:p-[12vh]" data-url><div id=search-modal class="flex flex-col w-full max-w-3xl min-h-0 mx-auto border rounded-md shadow-lg top-20 border-neutral-200 bg-neutral dark:border-neutral-700 dark:bg-neutral-800"><header class="relative z-10 flex items-center justify-between flex-none px-2"><form class="flex items-center flex-auto min-w-0"><div class="flex items-center justify-center w-8 h-8 text-neutral-400"><span class="relative inline-block align-text-bottom px-1 icon"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="search" class="svg-inline--fa fa-search fa-w-16" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentcolor" d="M505 442.7 405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9.0 208 0S0 93.1.0 208s93.1 208 208 208c48.3.0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9.0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7.0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7.0 128 57.2 128 128 0 70.7-57.2 128-128 128z"/></svg></span></div><input type=search id=search-query class="flex flex-auto h-12 mx-1 bg-transparent appearance-none focus:outline-dotted focus:outline-2 focus:outline-transparent" placeholder=Search tabindex=0></form><button id=close-search-button class="flex items-center justify-center w-8 h-8 text-neutral-700 hover:text-primary-600 dark:text-neutral dark:hover:text-primary-400" title="Close (Esc)">
<span class="relative inline-block align-text-bottom px-1 icon"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><path fill="currentcolor" d="M310.6 361.4c12.5 12.5 12.5 32.75.0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3 54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75.0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75.0-45.25s32.75-12.5 45.25.0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25.0s12.5 32.75.0 45.25l-105.4 105.4L310.6 361.4z"/></svg></span></button></header><section class="flex-auto px-2 overflow-auto"><ul id=search-results></ul></section></div></div></div></body></html>