---
title: "Weekly news #12"
date: 2021-03-21
tags: [weekly]
---

## <i class="fas fa-bullhorn"></i> News

* [Git 2.31](https://github.blog/2021-03-15-highlights-from-git-2-31/)
* [CutiePi tablet](https://cutiepi.io/)

---

## <i class="fas fa-dna"></i> Science/Bioinformatics

### [Modelling human blastocysts by reprogramming fibroblasts into iBlastoids](https://www.nature.com/articles/s41586-021-03372-y)

_Liu et al., Nature (2021)_

<a href="#" class="badge badge-primary">#blastocyst</a>
<a href="#" class="badge badge-primary">#human</a>
<a href="#" class="badge badge-primary">#scrna-seq</a>

_Human pluripotent and trophoblast stem cells have been essential alternatives to blastocysts for understanding early human development1,2,3,4. However, these simple culture systems lack the complexity to adequately model the spatiotemporal cellular and molecular dynamics that occur during early embryonic development. Here we describe the reprogramming of fibroblasts into in vitro three-dimensional models of the human blastocyst, termed iBlastoids. Characterization of iBlastoids shows that they model the overall architecture of blastocysts, presenting an inner cell mass-like structure, with epiblast- and primitive endoderm-like cells, a blastocoel-like cavity and a trophectoderm-like outer layer of cells. Single-cell transcriptomics further confirmed the presence of epiblast-, primitive endoderm-, and trophectoderm-like cells. Moreover, iBlastoids can give rise to pluripotent and trophoblast stem cells and are capable of modelling, in vitro, several aspects of the early stage of implantation. In summary, we have developed a scalable and tractable system to model human blastocyst biology; we envision that this will facilitate the study of early human development and the effects of gene mutations and toxins during early embryogenesis, as well as aiding in the development of new therapies associated with in vitro fertilization._

### [Blastocyst-like structures generated from human pluripotent stem cells](https://www.nature.com/articles/s41586-021-03356-y)

_Yu et al., Nature (2021)_

<a href="#" class="badge badge-primary">#blastocyst</a>
<a href="#" class="badge badge-primary">#human</a>
<a href="#" class="badge badge-primary">#scrna-seq</a>

_Limited access to embryos has hampered the study of human embryogenesis and disorders that occur during early pregnancy. Human pluripotent stem cells provide an alternative means to study human development in a dish1,2,3,4,5,6,7. Recent advances in partial embryo models derived from human pluripotent stem cells have enabled human development to be examined at early post-implantation stages8,9,10,11,12,13,14. However, models of the pre-implantation human blastocyst are lacking. Starting from naive human pluripotent stem cells, here we developed an effective three-dimensional culture strategy with successive lineage differentiation and self-organization to generate blastocyst-like structures in vitro. These structures—which we term ‘human blastoids’—resemble human blastocysts in terms of their morphology, size, cell number, and composition and allocation of different cell lineages. Single-cell RNA-sequencing analyses also reveal the transcriptomic similarity of blastoids to blastocysts. Human blastoids are amenable to embryonic and extra-embryonic stem cell derivation and can further develop into peri-implantation embryo-like structures in vitro. Using chemical perturbations, we show that specific isozymes of protein kinase C have a critical function in the formation of the blastoid cavity. Human blastoids provide a readily accessible, scalable, versatile and perturbable alternative to blastocysts for studying early human development, understanding early pregnancy loss and gaining insights into early developmental defects._

### [Checklist for Artificial Intelligence in Medical Imaging (CLAIM): A Guide for Authors and Reviewers](https://pubs.rsna.org/doi/10.1148/ryai.2020200029)

_Mongan el at., Rad. Artif. Intel (2021)_

<a href="#" class="badge badge-primary">#checklist</a>
<a href="#" class="badge badge-primary">#ai</a>
<a href="#" class="badge badge-primary">#images</a>

### [An automated framework for efficiently designing deep convolutional neural networks in genomics](https://www.nature.com/articles/s42256-021-00316-z)

_Zhang et al., Nat Mach Intell (2021)_

<a href="#" class="badge badge-primary">#genomics</a>
<a href="#" class="badge badge-primary">#ai</a>
<a href="#" class="badge badge-primary">#amber</a>

_Convolutional neural networks (CNNs) have become a standard for analysis of biological sequences. Tuning of network architectures is essential for a CNN’s performance, yet it requires substantial knowledge of machine learning and commitment of time and effort. This process thus imposes a major barrier to broad and effective application of modern deep learning in genomics. Here we present Automated Modelling for Biological Evidence-based Research (AMBER), a fully automated framework to efficiently design and apply CNNs for genomic sequences. AMBER designs optimal models for user-specified biological questions through the state-of-the-art neural architecture search (NAS). We applied AMBER to the task of modelling genomic regulatory features and demonstrated that the predictions of the AMBER-designed model are significantly more accurate than the equivalent baseline non-NAS models and match or even exceed published expert-designed models. Interpretation of AMBER architecture search revealed its design principles of utilizing the full space of computational operations for accurately modelling genomic sequences. Furthermore, we illustrated the use of AMBER to accurately discover functional genomic variants in allele-specific binding and disease heritability enrichment. AMBER provides an efficient automated method for designing accurate deep learning models in genomics._

### [Machine learning for deciphering cell heterogeneity and gene regulation](https://www.nature.com/articles/s43588-021-00038-7)

_Scherer et al., Nat Comput Sci (2021)_

<a href="#" class="badge badge-primary">#ai</a>
<a href="#" class="badge badge-primary">#epigenetics</a>

_Epigenetics studies inheritable and reversible modifications of DNA that allow cells to control gene expression throughout their development and in response to environmental conditions. In computational epigenomics, machine learning is applied to study various epigenetic mechanisms genome wide. Its aim is to expand our understanding of cell differentiation, that is their specialization, in health and disease. Thus far, most efforts focus on understanding the functional encoding of the genome and on unraveling cell-type heterogeneity. Here, we provide an overview of state-of-the-art computational methods and their underlying statistical concepts, which range from matrix factorization and regularized linear regression to deep learning methods. We further show how the rise of single-cell technology leads to new computational challenges and creates opportunities to further our understanding of epigenetic regulation._

---

## <i class="fas fa-toolbox"></i> Tools

### [manutamminen/blaster](https://github.com/manutamminen/blaster)

BLAST-like algorithm for R

### [mathiasbynens/dotfiles](https://github.com/mathiasbynens/dotfiles)

Inspiration for customizing your dot files. You can use them to setup your environment.

### [xbarapp](https://xbarapp.com/)

Formerly known as BitBar

### [Focalboard](https://www.focalboard.com/)

Focalboard is an open source, self-hosted alternative to Trello, Notion, and Asana.

---

## <i class="fas fa-graduation-cap"></i> Guides and Tutorials

### [Understanding Best Practice Python Tooling by Comparing Popular Project Templates](https://betterprogramming.pub/understanding-best-practice-python-tooling-by-comparing-popular-project-templates-6eba49229106)

A look at well-known and up-and-coming Python tools

### [Data Science in Julia for Hackers](https://datasciencejuliahackers.com/)

This book is written by Federico Carrone, Herman Obst Demaestri and Mariano Nicolini.

### [5 Tips for Writing Clean R Code – Leave Your Code Reviewer Commentless](https://appsilon.com/write-clean-r-code/)

Nice tips for writing R code.

### [Getting started with … Rust](https://stackoverflow.blog/2021/03/15/getting-started-with-rust/)

In this series, we look at the most loved languages according to the Stack Overflow developer survey,
the spread and use cases for each of them and collect some essential links on how to get into them. First up: Rust.

## [Learn LaTeX](https://www.learnlatex.org/en/)

Interactive learning resource for learning LaTeX.

---

## <i class="fas fa-rss"></i> Others

### [](https://github.com/mathiasbynens/dotfiles/blob/master/.macos)

### [Massively Parallel Graph Computation: From Theory to Practice](https://ai.googleblog.com/2021/03/massively-parallel-graph-computation.html)

### [Deep Dive into Docker Internals - Union Filesystem](https://martinheinz.dev/blog/44)

### [Which programming languages pay the most? I made my own salary charts...](https://stribny.name/blog/salaries-by-language/)

### [Principal Component Analysis | MIT Computational Thinking | Spring 2021 | Lecture 8](https://www.youtube.com/watch?v=iuKrM_NzxCk)
