---
title: "Weekly news #11"
date: 2021-03-14
tags: [weekly]
---

## <i class="fas fa-bullhorn"></i> News

* [htsget-rs in GSoC 2021](https://twitter.com/braincode/status/1369851695442751492)
* [7-zip officialy on Linux](https://www.bleepingcomputer.com/news/software/7-zip-developer-releases-the-first-official-linux-version/)

---

## <i class="fas fa-dna"></i> Science/Bioinformatics

### [R.ROSETTA: an interpretable machine learning framework](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04049-z)

_Garbulowski et al., BMC Bioinformatics (2021)_

<a href="#" class="badge badge-primary">#machine-learning</a>
<a href="#" class="badge badge-primary">#roseta</a>

_We present the R.ROSETTA package, which is an R wrapper of ROSETTA framework. The original ROSETTA functions have been improved and adapted to the R programming environment. The package allows for building and analyzing non-linear interpretable machine learning models. R.ROSETTA gathers combinatorial statistics via rule-based modelling for accessible and transparent results, well-suited for adoption within the greater scientific community. The package also provides statistics and visualization tools that facilitate minimization of analysis bias and noise. The R.ROSETTA package is freely available at https://github.com/komorowskilab/R.ROSETTA. To illustrate the usage of the package, we applied it to a transcriptome dataset from an autism caseâ€“control study. Our tool provided hypotheses for potential co-predictive mechanisms among features that discerned phenotype classes. These co-predictors represented neurodevelopmental and autism-related genes._

### [Deep learning-based enhancement of epigenomics data with AtacWorks](https://www.nature.com/articles/s41467-021-21765-5)

_Chiang et al., Nat Commun (2021)_

<a href="#" class="badge badge-primary">#atac-seq</a>
<a href="#" class="badge badge-primary">#machine-learning</a>

_ATAC-seq is a widely-applied assay used to measure genome-wide chromatin accessibility; however, its ability to detect active regulatory regions can depend on the depth of sequencing coverage and the signal-to-noise ratio. Here we introduce AtacWorks, a deep learning toolkit to denoise sequencing coverage and identify regulatory peaks at base-pair resolution from low cell count, low-coverage, or low-quality ATAC-seq data. Models trained by AtacWorks can detect peaks from cell types not seen in the training data, and are generalizable across diverse sample preparations and experimental platforms. We demonstrate that AtacWorks enhances the sensitivity of single-cell experiments by producing results on par with those of conventional methods using ~10 times as many cells, and further show that this framework can be adapted to enable cross-modality inference of protein-DNA interactions. Finally, we establish that AtacWorks can enable new biological discoveries by identifying active regulatory regions associated with lineage priming in rare subpopulations of hematopoietic stem cells._

### [Deep learning-based enhancement of epigenomics data with AtacWorks](https://www.nature.com/articles/s41467-021-21765-5)

_Lal and Chiang et al., Nat Commun (2021)_

<a href="#" class="badge badge-primary">#atac-seq</a>
<a href="#" class="badge badge-primary">#denoise</a>

_ATAC-seq is a widely-applied assay used to measure genome-wide chromatin accessibility; however, its ability to detect active regulatory regions can depend on the depth of sequencing coverage and the signal-to-noise ratio. Here we introduce AtacWorks, a deep learning toolkit to denoise sequencing coverage and identify regulatory peaks at base-pair resolution from low cell count, low-coverage, or low-quality ATAC-seq data. Models trained by AtacWorks can detect peaks from cell types not seen in the training data, and are generalizable across diverse sample preparations and experimental platforms. We demonstrate that AtacWorks enhances the sensitivity of single-cell experiments by producing results on par with those of conventional methods using ~10 times as many cells, and further show that this framework can be adapted to enable cross-modality inference of protein-DNA interactions. Finally, we establish that AtacWorks can enable new biological discoveries by identifying active regulatory regions associated with lineage priming in rare subpopulations of hematopoietic stem cells._

### [Pattern discovery and disentanglement on relational datasets](https://www.nature.com/articles/s41598-021-84869-4)

_Wong et al., Sci Rep (2021)_

<a href="#" class="badge badge-primary">#pattern-discovery</a>
<a href="#" class="badge badge-primary">#knowledge-discovery</a>

_Machine Learning has made impressive advances in many applications akin to human cognition for discernment. However, success has been limited in the areas of relational datasets, particularly for data with low volume, imbalanced groups, and mislabeled cases, with outputs that typically lack transparency and interpretability. The difficulties arise from the subtle overlapping and entanglement of functional and statistical relations at the source level. Hence, we have developed Pattern Discovery and Disentanglement System (PDD), which is able to discover explicit patterns from the data with various sizes, imbalanced groups, and screen out anomalies. We present herein four case studies on biomedical datasets to substantiate the efficacy of PDD. It improves prediction accuracy and facilitates transparent interpretation of discovered knowledge in an explicit representation framework PDD Knowledge Base that links the sources, the patterns, and individual patients. Hence, PDD promises broad and ground-breaking applications in genomic and biomedical machine learning._

### [Chunkflow: hybrid cloud processing of large 3D images by convolutional nets](https://www.nature.com/articles/s41592-021-01088-5)

_Wu et al., Nat Methods (2021)_

<a href="#" class="badge badge-primary">#image-processing</a>
<a href="#" class="badge badge-primary">#distributed-computation</a>

_Automated microscopes with both high resolution and large field of view are generating terascale and even petascale 3D images. A local cluster might not have enough computational resources to process them in reasonable time, but public cloud platforms can provide computational resources on demand. Convolutional networks have become the state-of-the-art approach for 3D biological image analysis1,2, and cloud processing by 3D convolutional nets has been used for processing independent small image stacks3,4,5. However, cloud computing tools to perform distributed processing of terascale or petascale 3D images by convolutional nets are lacking. Here, we report chunkflow, a framework for distributing computational tasks over both cloud and local computational resources, including both GPUs and CPUs with multiple deep-learning framework back ends, to maximize efficiency, increase flexibility and reduce cost._

---

## <i class="far fa-keyboard"></i> Programming

### [R Cheat Sheets](https://rviews.rstudio.com/2021/03/10/rstudio-open-source-resorurces/)

Resource of various cheat sheets for working with R

---

## <i class="fas fa-toolbox"></i> Tools

### [omnisci/omniscidb](https://github.com/omnisci/omniscidb)

OmniSci is the world's fastest open source SQL engine, equally powerful at the
heart of the OmniSci platform as it is accelerating third-party analytic apps.

### [rbreaves/kinto](https://github.com/rbreaves/kinto/)

Linux & Windows with Mac-style shortcut keys.

### [Clone Wars - Open source clones of popular sites](https://gourav.io/clone-wars)

70+ open-source clones of popular sites like Airbnb, Amazon, Instagram, Netflix, Tiktok, Spotify, Trello, Whatsapp, Youtube, etc.
List contains source code, demo links, tech stack, and, GitHub stars count. Great for learning purpose!

---

## <i class="fas fa-graduation-cap"></i> Guides and Tutorials

### [Data Science for Psychologists](https://bookdown.org/hneth/ds4psy/)

This book provides an introduction to data science that is tailored to the needs of psychologists,
but is also suitable for students of the humanities and other biological or social sciences.

### [OH MY GIT](https://ohmygit.org/)

An open source game about learning Git!

### [Docker Security Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html)

The aim of this cheat sheet is to provide an easy to use list of common security mistakes
and good practices that will help you secure your Docker containers.

### [How to Backup Docker Containers? Docker Container Backup Methods](https://www.baculasystems.com/blog/docker-backup-containers/)

Guide on how to backup docker container: `commin` -> `save` -> `tag` -> `push`.

### [Clustering similar spatial patterns](https://www.r-bloggers.com/2021/03/clustering-similar-spatial-patterns/)

### [Code Review Checklist R Code Edition Top 3](https://www.r-bloggers.com/2021/03/code-review-checklist-r-code-edition-top-3/)

---

## <i class="fas fa-rss"></i> Others

### [authelia/authelia](https://github.com/authelia/authelia)

The Single Sign-On Multi-Factor portal for web apps

### [Fitbit is doomed: Here's why everything Google buys turns to garbage](https://www.zdnet.com/article/fitbit-is-doomed-why-everything-google-buys-turns-to-garbage/)

---

## <i class="fas fa-file-alt"></i> Honorable mentions

### [The Polygenic Score Catalog as an open database for reproducibility and systematic evaluation](https://www.nature.com/articles/s41588-021-00783-5)

### [Prioritizing non-coding regions based on human genomic constraint and sequence context with deep learning](https://www.nature.com/articles/s41467-021-21790-4)

### [Deep learning-based point-scanning super-resolution imaging](https://www.nature.com/articles/s41592-021-01080-z)

### [Integration and transfer learning of single-cell transcriptomes via cFIT](https://www.pnas.org/content/118/10/e2024383118)